### DISCUSSION TITLE
Is it possible to update rows based on their current values via the client?

### DISCUSSION THREAD

**Discussion by officiallymarky** [supabase/supabase#2856] [Questions]
Ideally I want to use the Python client to do this.

I would like to update the **inventory** table and set the **balance** field for user xyz to be **balance + 5**.  I know this is possible with Postgres, is there anyway to do this through either of the clients or via the API or is the only way to do this is directly through Postgres queries?

I am looking to update a record without having to query for the current value and do the update atomically.

**Comment by darora** [comment]
Hey @officiallymarky - one way you could do this would be to create a stored procedure in your database, and then invoke it through the postgrest python client.

https://postgrest.org/en/v8.0/api.html#stored-procedures
https://github.com/supabase/postgrest-py/blob/v0.3.2/postgrest_py/client.py#L61-L66
